api/comment_threads.rb

```diff
@@ -54,10 +54,15 @@
   filter_blocked_content params["body"]
   thread.update_attributes(params.slice(*%w[title body pinned closed commentable_id group_id thread_type]))
 
+  if params["user_id"] and value_to_boolean(params["read"])
+    user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
+    user.mark_as_read(thread) if user
```

Comment:
if we have no user at this point, wouldn't we want some sort of exception or error logged to indicate something unexpected is happening?